.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,33 @@
+name: Publish Docker image
+on:
+  release:
+    types: [published]
+  push:
+    branches:
+      - master
+      - dev
+  pull_request:
+    branches: [dev]
+
+jobs:
+  push_to_registries:
+    name: Push Docker Image to DockerHub and Github Packages
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+      - name: Push to Docker Hub
```

Comment:
Perhaps it would be good to consider tagging images with **branch** or **sha_short** and have published releases push an image tagged as **latest** and **version**? This would make the package releases cleaner.

Reply:
Also, is it necessary to push to Docker Hub?

Reply:
Migration to Docker isn't complete yet. See #23 for details on how it was setup and the limitations of its current implementation. Additional work needs to be done before the fully containerized workflow can be used. The current implementation (local development, ci/cd, deployment, etc) is 100% native right now. We've prioritized feature development over moving to docker since its a low priority item.

Eventually, we will _not_ publish to either docker hub or GitHub packages. We would only publish to Google Cloud Registry since the container image would contain secrets that we want to keep private. Furthermore, integrations between GCP Registry and Cloud Build / Kubernetes / Cloud Run / etc are very well designed so we wouldn't need to have images saved in multiple places. This `.yml` file was more of an exercise to see what container workflows are feasible and put something into place that we can swap out later.